packages/edit-post/src/components/sidebar/plugin-pre-publish-panel/index.js

```diff
@@ -76,13 +58,28 @@ const PluginPrePublishPanelFill = ( {
  *
  * @return {WPComponent} The component to be rendered.
  */
-const PluginPrePublishPanel = compose(
-	withPluginContext( ( context, ownProps ) => {
-		return {
-			icon: ownProps.icon || context.icon,
-		};
-	} )
-)( PluginPrePublishPanelFill );
+const PluginPrePublishPanel = ( {
+	children,
+	className,
+	title,
+	initialOpen = false,
+	icon,
+} ) => {
+	const context = usePluginContext();
```

Comment:
Optional: since we only use `icon` from the `context`, we might as well destructure it from `context`:

```suggestion
	const { icon: pluginIcon } = usePluginContext();
```